.github/workflows/buildandpublish.yml

```diff
@@ -0,0 +1,31 @@
+name: Build and publish Ansible
+on:
+  push:
```

Comment:
Personally, I like having the `workflow_dispatch` trigger. With it, you could have a form with an input for the desired version, right in the GitHub UI. Then, the workflow could publish both to PyPI and push a tag to GitHub (and maybe a GitHub Release). If you push a tag yourself, this would mean a possibility of a tag appearing in the repo but the actual building and publishing failing.